### PR TITLE
Fix incorrect test and clean up edge case for Vector Index

### LINKED ISSUE #1249: [Feature]: Cover edge case in VectorIndex get refactor
### What would you like to see added to HDMF?

https://github.com/hdmf-dev/hdmf/pull/1248

### What solution would you like?

TBD

### Do you have any interest in helping implement the feature?

Yes.

### PR DESCRIPTION
**Pull request by mavaylon1**
## Motivation

Fix #1249 
## How to test the behavior?
```
Show how to reproduce the new behavior (can be a bug fix or a new feature)
```

## Checklist

- [x] Did you update `CHANGELOG.md` with your changes?
- [x] Does the PR clearly describe the problem and the solution?
- [x] Have you reviewed our [Contributing Guide](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst)?
- [x] Does the PR use "Fix #XXX" notation to tell GitHub to close the relevant issue numbered XXX when the PR is merged?

### CODE REVIEW

**Comment by mavaylon1** [comment]
Review Notes: This should be ready to go, but possible coverage may need an extra test to get it covered 100%

File: tests/unit/common/test_linkedtables.py [hunk 1]
```diff
@@ -445,7 +445,7 @@ def test_to_hierarchical_dataframe_indexed_dtr_on_last_level(self):
         p1 = DynamicTable(name='parent_table', description='parent_table',
                           columns=[VectorData(name='p1', description='p1', data=np.arange(3)), dtr_p1, vi_dtr_p1])
         # Super-parent table
-        dtr_sp = DynamicTableRegion(name='sl1', description='sl1', data=np.arange(4), table=p1)
+        dtr_sp = DynamicTableRegion(name='sl1', description='sl1', data=np.arange(3), table=p1)
```
Comment (rly) [inline tests/unit/common/test_linkedtables.py:448]: The previous data range was invalid because the target table only has 3 rows. This was causing an IndexError in `VectorIndex.get`.

Comment (rly) [reply tests/unit/common/test_linkedtables.py:448]: @oruebel since you set up this test originally, please confirm that I am not missing something

**Review by oruebel — APPROVED** [review]
